src/v/kafka/server/server.cc

```diff
@@ -366,6 +366,11 @@ ss::future<response_ptr> heartbeat_handler::handle(
     request.decode(ctx.reader(), ctx.header().version);
     log_request(ctx.header(), request);
 
+    if (ctx.recovery_mode_enabled()) {
```

Comment:
this and offset commit are performance critical code paths please wrap with `unlikely`